provisioner/shell/provisioner.go

```diff
@@ -202,20 +210,38 @@ func (p *Provisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packe
 		scripts = append(scripts, tf.Name())
 
 		// Write our contents to it
-		writer := bufio.NewWriter(tf)
-		_, _ = writer.WriteString(fmt.Sprintf("#!%s\n", p.config.InlineShebang))
+		fileWriter := bufio.NewWriter(tf)
+		commandBuffer := &bytes.Buffer{}
 		for _, command := range p.config.Inline {
 			p.config.ctx.Data = generatedData
 			command, err := interpolate.Render(command, &p.config.ctx)
 			if err != nil {
 				return fmt.Errorf("Error interpolating Inline: %s", err)
 			}
-			if _, err := writer.WriteString(command + "\n"); err != nil {
+			if _, err := commandBuffer.WriteString(command + "\n"); err != nil {
 				return fmt.Errorf("Error preparing shell script: %s", err)
 			}
 		}
 
-		if err := writer.Flush(); err != nil {
+		firstLine, err := commandBuffer.ReadString('\n')
```

Comment:
You don't need to read the first line I'd think, instead you can probably use `String()` from the buffer you created, if it's a `strings.Builder`, the operation is virtually free, so no need to perform this extra computation.